### DISCUSSION TITLE
pdfplumber extract no chars, but lines is correct. Problems occurs in remote server. In local computer, everything works fine.

### DISCUSSION THREAD

**Discussion by smilealvin92** [jsvine/pdfplumber#478] [Q&A]
pdfplumber extract no chars, but lines is correct. Problems occurs in remote server. In local computer, everything works fine. In the first place, I guess its the fonts which cause the problem. But after I install the fonts on remote server, the problem still occurs.

**Comment by samkit-jain** [comment]
Hi @smilealvin92 Can you please provide more information? What environment are you running on? What is the pdfplumber version? What is the expected output and what is the actual output? The PDF file (redacting any sensitive information).

**Reply by smilealvin92** [reply]
sorry, I was very upset. But I have solve the problem now. let me explain it. The truth is, when we install pdfplumber==0.5.24 in remote server, it will automatically install pdfminer.six==20200517, however, this 0517 is not working in my project, which can not extract any chars. In my local machine, I check pdfminer.six is 20201018, which may be installed by coincidence. I change the version of pdfminer.six in remote server to 20201018 manually, my project works fine. which surprises me is that the pip report error when I switch the version to 2020108, it says that 20201018 is not compatible with pdfplumber of 0.5.24.

**Reply by smilealvin92** [reply]
I am just using  "pdf = pdfplumber.open(pdf)" this simple code. I hope it can extract the chars in pdf.

**Reply by smilealvin92** [reply]
I doubt anybody has encountered the same problem as me. So I post my solution here, hope it helps.

**Comment by smilealvin92** [comment]
![2021-07-21 16-07-56 的屏幕截图](https://user-images.githubusercontent.com/48876213/126454349-953689b4-efd3-4199-9e43-eeda326d00d2.png)
